### PR TITLE
Added type definition

### PR DESCRIPTION
**Pull request by kotonefami**
Refer to the API reference in README.md and add the TypeScript type definition file.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,8 +23,8 @@
 	"scripts": {
 		"build": "yarn dist && vite build --config demo/vite.config.js",
 		"dev": "concurrently \"yarn watch\" \"vite --config demo/vite.config.js --port 3000\"",
-		"watch": "microbundle watch --target web --globals three=THREE --external three",
-		"dist": "microbundle --target web --globals three=THREE --external three",
+		"watch": "microbundle watch --target web --globals three=THREE --external three --no-generateTypes && node scripts/copy-definition.mjs",
+		"dist": "microbundle --target web --globals three=THREE --external three --no-generateTypes && node scripts/copy-definition.mjs",
```
Comment (donmccurdy) [inline package.json:27]: I seem to be getting the same result without `--no-generateTypes`, is that needed for something?

File: src/three-pathfinding.d.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
+import type { BufferGeometry, Object3D, Vector3 } from "three";
+
+declare module "three-pathfinding" {
```
Comment (donmccurdy) [inline src/three-pathfinding.d.ts:3]: Is the 'declare module'  block required? In other some packages I maintain — written in TypeScript, and compiled with Microbundle — the generated type definitions don't have this block, so I just want to make sure I understand the need, if any. :)

**Review by donmccurdy — COMMENTED** [review]
Thank you @kotonefami! Just a couple questions, and then happy to merge this.

File: src/three-pathfinding.d.ts (hunk 2 above)
Comment (kotonefami) [reply src/three-pathfinding.d.ts]: This was not necessary for the 'declare module' block. I am very sorry about that.

File: package.json (hunk 1 above)
Comment (kotonefami) [reply package.json:27]: In my environment, running watch or dist without `--no-generateTypes` produced index.d.ts like this

```ts
import { Pathfinding } from "./Pathfinding";
import { PathfindingHelper } from "./PathfindingHelper";
export { Pathfinding, PathfindingHelper };
```

So I turned off the default type generation, will this file not be uploaded to npm?

Comment (donmccurdy) [reply package.json:27]: I see, thank you!

**Comment by donmccurdy** [comment]
Thanks for this, and sorry for the slow responses. :)

**Comment by donmccurdy** [comment]
Published to v1.3.

**Comment by kotonefami** [comment]
Thank you for your merging!

**Pull request merged**